core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkspaceManagerResourceMonitorRecordComponent.scala

```diff
@@ -148,7 +153,19 @@ final case class WorkspaceManagerResourceMonitorRecord(
   userEmail: Option[String],
   createdTime: Timestamp,
   args: Option[Map[String, String]] = None
-)
+) {
+
+  def retryOrTimeout(onTimeout: () => Future[Any] = () => Future.successful())(implicit
+    executionContext: ExecutionContext
+  ): Future[JobStatus] = {
+    val expireTime = Instant.ofEpochMilli(createdTime.getTime).plus(1, ChronoUnit.DAYS)
+    if (expireTime.isBefore(Instant.now())) {
+      onTimeout().map(_ => Complete)
```

Comment:
Are you storing the last encountered error anywhere?

Reply:
Oh, I see… it is passed in the lambda.

Reply:
Yes, because the actual process is going to differ depending on the job.